source/features/useful-not-found-page.tsx

```diff
@@ -29,71 +27,247 @@ async function checkAnchor(anchor: HTMLAnchorElement): Promise<void> {
 
 function parseCurrentURL(): string[] {
 	const parts = getCleanPathname().split('/');
-	if (parts[2] === 'blob') { // Blob URLs are never useful
+	// Blob URLs are never useful
+	if (parts[2] === 'blob') {
 		parts[2] = 'tree';
 	}
 
 	return parts;
 }
 
-// If the resource was deleted, link to the commit history
-async function addCommitHistoryLink(bar: Element): Promise<void> {
-	const parts = parseCurrentURL();
-	parts[2] = 'commits';
-	const url = '/' + parts.join('/');
-	if (await is404(location.origin + url)) {
+async function getUrlToFileOnDefaultBranch(): Promise<string | undefined> {
+	// Get the current branch
+	const url = new GitHubURL(location.href);
+	const currentBranch = url.branch;
+
+	if (!currentBranch) {
 		return;
 	}
 
-	bar.after(
-		<p className="container mt-4 text-center">
-			See also the {getType()}’s <a href={url}>commit history</a>.
-		</p>,
+	const defaultBranch = await getDefaultBranch();
+	if (currentBranch === defaultBranch) {
+		return;
+	}
+
+	// Change branch
+	url.assign({
+		branch: defaultBranch
+	})
+	// Check if that path exists
+	if (await is404(url.toString())) {
+		return;
+	}
+
+	// If it does, return it
+	return url.toString();
+}
+
+async function getLastCommitForFile(
```

Comment:
The name does not match the return value. Ensure that functions do one thing and their names match what they return